main/modes/mode_diceroller.c

```diff
@@ -0,0 +1,635 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <math.h>
+
+#include "cndraw.h"
+#include "esp_timer.h"
+#include "swadgeMode.h"
+#include "swadge_esp32.h"
+#include "led_util.h" // leds
+#include "meleeMenu.h"
+#include "mode_main_menu.h"
+#include "nvs_manager.h"
+#include "bresenham.h"
+#include "display.h"
+#include "embeddednf.h"
+#include "embeddedout.h"
+#include "musical_buzzer.h"
+#include "settingsManager.h"
+#include "linked_list.h"
+//#include "bootloader_random.h"
+#include "esp_random.h"
+
+#include "swadge_util.h"
+#include "text_entry.h"
+
+#include "mode_diceroller.h"
+
+
+//TODO: Add Doxygen Blocks
+//TODO: Add Cosmetic Features of Colored background/foreground objects. Add Smoother animations. Add LED interaction.
+
+const double pi = 3.14159265359;

```

Comment:
This is in `math.h` as `M_PI`

Reply:
Fixed inability to use M_PI by defining _USE_MATH_DEFINES